dataplicity/portforward.py

```diff
@@ -24,12 +24,19 @@ class Connection(threading.Thread):
     # Max to read at-a-time
     BUFFER_SIZE = 1024 * 32
 
-    def __init__(self, service, connection_id, channel):
+    def __init__(self, close_event, connection_id, channel,
+                 service=None, host_port=None):
         """Initialize the connection, set up callbacks."""
         super(Connection, self).__init__()
-        self._service = weakref.ref(service)
+        self._close_event = close_event
```

Comment:
Why are you passing in the close event now?

Reply:
huh. that is a bloody good question :/ I'll have to re-read this code. This is probably some old code from the previous implementation

Reply:
hmm ok so after re-inspecting the code I see that I am passing it due to this line:

```            while not self.close_event.is_set():
```

the thing is, the original implementation of `close_event` property was:

```
return self.service.close_event
```

but because the Connection thread will be used even when service is None, I had to pass the event explicitely and change the implementation of the property into:

```
        return self._close_event
```

for backwards compatibility